### ISSUE TITLE
catt doesn't allow to select subtitles inside the container

### ISSUE THREAD

**Issue by guiniol** [skorokithakis/catt#117]
The `--subtitle` option looks for either an URL or a local file. There is no way to specify a track inside the container.
`catt --info` does show that the subtitle tracks present inside the container.
It would be nice if those could be selected.

Not sure how much work that would be. And in any case, thank you for `catt`. It's so much better than going through chromium or an app on my phone!

**Comment by theychx** [comment]
Thank you for this suggestion.
The problem is that the chromecast does not support the subrip (.srt) subtitle format, rendering such a feature not very useful (external subrip subtitles are working with ```catt``` because we are converting them to webvtt, the only common format supported by the chromecast).

**Comment by guiniol** [comment]
Ah. So basically, you would have to extract the subtitle and convert it to webvtt?
That does sound impractical...

**Comment by theychx** [comment]
Indeed

**Comment by skorokithakis** [comment]
This is a good suggestion, but I'm afraid there isn't much we can do about it now... Should we close the ticket, but keep it in mind in case it becomes easier in the future?

**Comment by theychx** [comment]
We could use ```ffmpeg``` to extract the subtitles. I'm not really keen on such a solution, unless a lot of users are asking for this. I think it makes sense to keep this ticket open, so people can chime in if they are so inclined.

**Comment by skorokithakis** [comment]
You can't count on `ffmpeg` being installed, though :/ Alright, let's keep this open.

**Comment by theychx** [comment]
> You can't count on ffmpeg being installed, though

No, but you can check if it's there.

**Comment by skorokithakis** [comment]
Of course. I meant I wouldn't want to add it as a dependency, but we can just say "this only works with ffmpeg".

**Comment by theychx** [comment]
By the way, I tried embedding a webvtt subtitle in a mkv container, and I could not get it to work. Maybe I should ask the kind pychromecast people, if it's even possible to use internal subs.

**Comment by skorokithakis** [comment]
That would be great to know, I don't recall having embedded subtitles ever work either.

**Comment by guiniol** [comment]
Feel free to do whatever with this issue. People will find it more easily if it's opened, but if you don't plan to work on it (which I can certainly understand since the problem is with the chromecast), then closing it would be the normal thing to do.

**Comment by theychx** [comment]
@skorokithakis There's really not any point to keeping this open anymore, as this does not seem to be an annoyance to many users.
I'm still not keen on going down the `ffmpeg` route (in general), and it is my understanding that you agree.

**Comment by skorokithakis** [comment]
Alas, I do agree. It sounds like it's too big a hassle, unfortunately.

**Comment by EtienneM** [comment]
For other people looking for that kind of feature, I created a small script which extracts the subtitles from a mkv file and call `catt cast` with the freshly extracted subtitles. I uploaded the script to a gist if anybody wants to use it: https://gist.github.com/EtienneM/499ed604be10191cb3c709497264d65e

**Comment by theychx** [comment]
@EtienneM 
Neat, thanks!

**Comment by akvadrako** [comment]
> @skorokithakis There's really not any point to keeping this open anymore, as this does not seem to be an annoyance to many users.

I find this odd. Most things I watch have embedded subtitles.

**Comment by skorokithakis** [comment]
Actually, I agree, most things I watch have embedded subtitles too. It would be great if we could support them, even if it were really hacky and we had to say "you need ffmpeg for this". I kind of agree with @theychx for closing this too, as we'd basically need to download a remote file once, extract the subtitles, throw it away, start the local server so we can serve the subtitles, and keep the server open while the video plays.

Let's keep this open as an aspirational ticket, I might investigate some methods for extracting remote subs at some point, or someone who really wants this can open a PR and a ticket is a good way to indicate that we want this functionality.

**Comment by cmvanb** [comment]
I ran into this problem as well. Thank you to @EtienneM for the clever script that solved it!

I updated said script to work with latest ffmpeg: https://gist.github.com/cmvanb/8a288f2cf8f3081c5be8ea80dcc56eb8
